src/frontend/src/session.rs

```diff
@@ -436,6 +446,22 @@ impl FrontendEnv {
         });
         join_handles.push(join_handle);
 
+        let catalog_reader_for_report = catalog_reader.clone();
+        let join_handle = tokio::spawn(async move {
+            loop {
+                if let Err(err) =
+                    report_iceberg_metrics(&catalog_reader_for_report, &iceberg_stat_metrics).await
```

Comment:
This loop currently will be executed in every frontend node which is a bit redundant. Let's remove it from this PR.